src/common/CircleGameObject.cpp

```diff
@@ -1,18 +1,35 @@
 #include "CircleGameObject.hpp"
 
 
-CircleGameObject::CircleGameObject(unsigned int id, sf::Vector2f position, float size, sf::Color color)
-    : GameObject(id), mCircle(std::shared_ptr<Circle>(new Circle(size, position, color))) {}
+CircleGameObject::CircleGameObject(unsigned int id, sf::Vector2f position, float radius,
+    sf::Color color, std::shared_ptr<PhysicsController> pc)
+    : GameObject(id),
+      mCircle(std::shared_ptr<Circle>(new Circle(radius, position, color))),
+      mCircleRigidBody(pc->createCRB(radius, position)) {}
 
 std::shared_ptr<Circle> CircleGameObject::getCircle() {
     return mCircle;
 }
 
-std::vector<std::shared_ptr<Circle>> CircleGameObject::getCircles(
-    std::vector<std::shared_ptr<CircleGameObject>>& circle_game_objects) {
-    std::vector<std::shared_ptr<Circle>> circles;
-    std::transform(
-        circle_game_objects.begin(), circle_game_objects.end(), std::back_inserter(circles),
-        [](std::shared_ptr<CircleGameObject> cgo){return cgo->getCircle();});
-    return circles;
-};
+void CircleGameObject::setActive(bool is_active) {
+    mIsActive = is_active;
+    mCircleRigidBody->setActive(is_active);
+}
+
+void CircleGameObject::setVelocity(sf::Vector2f velocity) {
+    mCircleRigidBody->setVelocity(velocity);
+}
+
+void CircleGameObject::setRadius(float radius) {
+    mCircle->setRadius(radius);
+    mCircleRigidBody->setRadius(radius);
+}
+
+void CircleGameObject::setPosition(sf::Vector2f position) {
+    mCircle->setPosition(position);
+    mCircleRigidBody->setPosition(position);
+}
+
+void CircleGameObject::matchRigid() {
```

Comment:
when is this necessary? worth commenting since it's surprising at first glance. I'm assuming maybe in interpolation?

Reply:
So after the physics world updates the rigid bodies (applies velocity and corrects collision) the `CircleGameObject`s visual representation of the circle, needs to be updated to match the physical representation of the circle.

Substeps per step of GameController
> Updates game objects
> Takes a physics controller step
> Handles collision
> Updates game objects to match their rigid bodies (this step is a bit clunky, looking to improve)